contracts/receivables/BaseCreditPoolReceivable.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+import "hardhat/console.sol";
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "../BaseCreditPool.sol";
+
+contract BaseCreditPoolReceivable is ERC721, ERC721Enumerable, ERC721URIStorage, AccessControl {
```

Comment:
1. No need to inherit ERC721 here since ERC721URIStorage already inherits it. 

2. Is ERC721Enumerable indeed required? It will increase gas cost as you found in RN case.

Reply:
1. Thanks, will remove.
2. Yes, this one actually could benefit us because we'll need to render lists of a user's receivables on the dapp. Enumerability helps us with that. Gas costs shouldn't be a concern since we're launching this on Polygon.